ovh/data_okms_resource_gen.go

```diff
@@ -10,44 +8,47 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
 )
 
-func OkmsResourceDataSourceSchema(ctx context.Context) schema.Schema {
-	attrs := map[string]schema.Attribute{
-		"iam": schema.SingleNestedAttribute{
-			Attributes: map[string]schema.Attribute{
-				"display_name": schema.StringAttribute{
-					CustomType:          ovhtypes.TfStringType{},
-					Computed:            true,
-					Description:         "Resource display name",
-					MarkdownDescription: "Resource display name",
-				},
-				"id": schema.StringAttribute{
-					CustomType:          ovhtypes.TfStringType{},
-					Computed:            true,
-					Description:         "Unique identifier of the resource",
-					MarkdownDescription: "Unique identifier of the resource",
-				},
-				"tags": schema.MapAttribute{
-					CustomType:          ovhtypes.NewTfMapNestedType[ovhtypes.TfStringValue](ctx),
-					Computed:            true,
-					Description:         "Resource tags. Tags that were internally computed are prefixed with ovh:",
-					MarkdownDescription: "Resource tags. Tags that were internally computed are prefixed with ovh:",
-				},
-				"urn": schema.StringAttribute{
-					CustomType:          ovhtypes.TfStringType{},
-					Computed:            true,
-					Description:         "Unique resource name used in policies",
-					MarkdownDescription: "Unique resource name used in policies",
-				},
+func appendIamSchema(attrs map[string]schema.Attribute, ctx context.Context) {
```

Comment:
I'd rather move it to `type_iam.go` and also move all the `IamType` and `IamValue` stuff currently defined in `resource_vps_gen.go`, what do you think ?

Reply:
I've added a commit with these changes, but I see that `resource_vps_gen.go` is autogenerated, so maybe it's not worth unifying all the IAM types if it breaks the auto-generation of the vps resource.

Reply:
your change seems ok, even if `resource-vps-gen.go` is autogenerated we are planning a refactoring to move resources in several packages, so it will have to be modified anyway